packages/react-component-library/src/components/Autocomplete/hooks/useAutocomplete.ts

```diff
@@ -20,26 +20,39 @@ export function useAutocomplete(
   onIsOpenChange: (
     changes: UseComboboxStateChange<SelectChildWithStringType>
   ) => void
+  onSelectedItemChange: (
+    changes: UseComboboxStateChange<SelectChildWithStringType>
+  ) => void
 } {
   const [hasError, setHasError] = useState<boolean>(isInvalid)
   const inputRef = useRef<HTMLInputElement>(null)
-  const [items, setItems] = useState<SelectChildWithStringType[]>(children)
+  const [filterValue, setFilterValue] = useState<string>('')
+
+  const items = React.Children.toArray(children).filter((child) => {
```

Comment:
I agree with this approach store the text value in the state and filter in render